core/restore.php

```diff
@@ -0,0 +1,450 @@
+<?php
+/**
+ * @file
+ * Administrative page for restoring backup database and configuration files.
+ *
+ * Point your browser to "http://www.example.com/core/restore.php" and follow
+ * the instructions.
+ *
+ * If you are not logged in using either the site maintenance account or an
+ * account with the "Restore site backups" permission, you will need to
+ * modify the access check statement inside your settings.php file. After
+ * finishing the upgrade, be sure to open settings.php again, and change it
+ * back to its original state!
+ */
+
+/**
+ * Defines the root directory of the Backdrop installation.
+ */
+define('BACKDROP_ROOT', dirname(dirname($_SERVER['SCRIPT_FILENAME'])));
+
+// Change the directory to the Backdrop root.
+chdir(BACKDROP_ROOT);
+
+/**
+ * Global flag indicating that restore.php is being run.
+ *
+ * When this flag is set, various operations do not take place, such as invoking
+ * hook_init() and hook_exit(), css/js preprocessing, and translation.
+ */
+const MAINTENANCE_MODE = 'restore';
+
+/**
+ * Form constructor for the list of backups that can be restored.
+ */
+function restore_script_selection_form($form, &$form_state) {
+  $existing_backups = array();
+
+  // If there are no backups, display a message.
+  if (empty($existing_backups)) {
```

Comment:
$existing_backups is always empty - as it's defined as empty array right above. :wink: I don't actually get the comment above.

Currently this function isn't in use, yet, right? Is it supposed to get used in the future?